docs/contributing/reviewable-prs.md

```diff
@@ -1,79 +1,142 @@
-code reviewers to spot bugs, so if your PR does not follow the guidelines,
-reviewers will ask you to restructure it prior to going through the code.
-
-## Submit your pull request for review
-
-If you are new to Git, see our guide on [making a pull
-request][git-guide-make-pr] for detailed technical instructions on how to submit
-a pull request. When submitting your PR, you will need to:
-
-- Clearly describe the work you are submitting. Make sure the pull request
-  template is filled out correctly, and that all the relevant points on the
-  self-review checklist (if the repository has one) have been addressed. See the
-  [reviewable pull requests](../contributing/reviewable-prs.md) guide for
-  advice.
-
-- If you have a question that you expect to be resolved during the review
-  process, put it in a PR comment attached to a relevant part of the changes.
-  The review process will go a lot more smoothly if points of uncertainty
-  are explicitly laid out.
-
-- Make sure that the pull request passes all CI tests. You can sometimes
-  request initial feedback if there are open questions that will impact how
-  you update the tests. But in general, maintainers will wait for your PR to
-  pass tests before reviewing your work.
+A pull request (PR) is a presentation of your proposed changes to Zulip. Your aim
+should be to explain your changes as clearly as possible. This will help
+reviewers evaluate whether the proposed changes are correct, and address any
+open questions. Clear communication helps the whole Zulip project move more
+quickly by saving maintainers time when they review your code. It will also make
+a big difference for getting your work integrated without delay.
+
+You will go through the following steps to prepare your work for review. Each
+step is described in detail below, with links to additional resources:
+
+1. Write your code [with clarity in mind](#write-clear-code).
+1. [Organize your proposed changes](#organize-your-proposed-changes) into a
+   series of commits that tell the story of how the codebase will change.
+1. [Explain your changes](#explain-your-changes) in the description for your
+   pull request, including [screenshots](#demonstrating-visual-changes) for
+   visual changes.
+1. Carefully [review your own work](#review-your-own-work).
+1. [Submit your pull request](#submit-your-pull-request-for-review) for review.
+
+See the [pull request review process](../contributing/review-process.md) guide
+for a detailed overview of what happens once your pull request is submitted.
+
+## Write clear code
+
+When you write code, you should make sure that you understand *why it works* as
+intended. This is the foundation for being able to explain your proposed changes
+to others.
+
+Zulip’s coding philosophy is to focus relentlessly on making the codebase easy
+to understand and difficult to make dangerous mistakes. See the [code style
+guide](./code-style.md) for detailed guidance on how we approach this. Following
```

Comment:
I ended up reworking this paragraph to be a bit more expensive and ended up not linking to the code style guidelines with the revised language; I don't think of the zulip/zulip code style document as a particularly high quality page or any more important than the other pieces of our contributing guides, since so many of the more important code style things are implemented via linters and thus not written there.

See the diff below; we can keep revising on top of that, but I guess we don't need to do a list of links here.